// Copyright 2015 PKU-Cloud.
// Author: Chao Ma (mctt90@gmail.com)
//
// The message format is as follow:
// ------------------------------
// |  Header message:           |
// |   1. worker_id             |
// |   2. start_index           |
// |                            |
// |  list :                    |
// |   0.323                    |
// |   1.132                    |
// |   2.322                    |
// |   1.213                    |
// |   .                        |
// |   .                        |
// |   2.322                    |
// ------------------------------
package brook;

message HeadMessage {
    required uint32 worker_id = 1;
    required uint64 start_index = 2;
}

message DoubleMessage {
    required HeadMessage head = 1;
    repeated double list = 2;
}

message FloatMessage {
    required HeadMessage head = 1;
    repeated float list = 2;
}

message IntMessage {
    required HeadMessage head = 1;
    repeated int32 list = 2;
}

message Vector {
    repeated double value = 1;
}

message VectorMessage {
    required HeadMessage head = 1;
    repeated Vector list = 2;
}
